fabtests/common/shared.c

```diff
@@ -3059,6 +3066,44 @@ void ft_parse_hmem_opts(int op, char *optarg, struct ft_opts *opts)
 	}
 }
 
+void parse_opts_range (char* optarg)
+{
+       /* create a substring, take off prefix */
+       int rem_str_len = strlen (optarg) - 2;
+       char rem_str [rem_str_len];
+       strncpy (rem_str, &optarg[2], rem_str_len);
+
+       /* use string tokenizer */
+       const int MAX_TOKS = 3;
+       int inputs [MAX_TOKS];
+       const char* delim = ",";
+       char* token = strtok (rem_str, delim);
+       int j = 0;
+
+       while (token != NULL)
+       {
+	       inputs[j] = atoi (token);
+	       assert (j < MAX_TOKS);
+	       /* parse next token */
+	       token = strtok (NULL, delim);
+	       j++;
+       }
+       int start = inputs[0], inc = inputs[1], end = inputs[2];
```

Comment:
place all variable definitions at the top of the function.  Mingling definitions throughout a function makes them hard to find, making maintenance more difficult.

Reply:
Agreed, done.